controllers/posts_controller.js

```diff
@@ -5,16 +5,35 @@ const qs = require('querystring')
 
 module.exports.addPosts = (req, res) => {
     var userID = req.user.id
-    post.addPosts(req.body.description, req.body.url, req.body.tags, req.user.id, function (err, data) {
-        if(err)
-            return res.status(400).json({error: true, data: null})
-        user.linkPosts(userID, data, function (err, data) {
-            res.status(err ? 404 : 200).json({
-                error: err,
-                data: data
+    var url = req.body.url
```

Comment:
could have used pattern matching in the signature
`module.exports.addPosts = (({user: {id}, body: {url}}, res) => {`

or

`module.exports.addPosts = ({user, body}, res) => {` 
if you want to have access **body** or **user** properties of the *request*